Node client sparse array length can overflow before ERR_TOO_MUCH_DATA

## Summary

The Node native binding reads a JavaScript array length as `u32` and multiplies it by the event size before checking `constants.message_body_size_max`.

Sparse JavaScript arrays can have very large lengths without allocating corresponding elements. Passing such an array to a request method can make the native binding trap on integer overflow before returning `ERR_TOO_MUCH_DATA`.

## Affected Code

- `src/clients/node/node.zig`
- `src/clients/node/src/translate.zig`
- `src/clients/node/src/index.ts`

The relevant check multiplies before bounding:

```zig
const array_length: u32 = try translate.array_length(env, array);

if (array_length * @sizeOf(Event) > constants.message_body_size_max) {
    return request_error(env, .ERR_TOO_MUCH_DATA);
}
```

`translate.array_length()` returns the N-API array length as `u32`:

```zig
pub fn array_length(env: c.napi_env, array: c.napi_value) !u32 {
    var length: u32 = undefined;
    assert(c.napi_get_array_length(env, array, &length) == c.napi_ok);
    return length;
}
```

## Reproduction

```js
const tb = require("./src/clients/node/dist/index.js");

const client = tb.createClient({
  cluster_id: 0n,
  replica_addresses: [3000],
});

const batch = [];
batch.length = 0xffffffff;

client.createAccounts(batch);
```

Observed result:

```text
panic: integer overflow
src/clients/node/node.zig:281:30: in request
            if (array_length * @sizeOf(Event) > constants.message_body_size_max) {
                             ^
```

## Expected Behavior

The request should fail with `RequestError(ERR_TOO_MUCH_DATA)`.

## Actual Behavior

The native binding panics on integer overflow before it reaches the too-much-data branch.

## Impact

Any caller that can pass an arbitrary JavaScript batch array to the public Node client can abort the Node process with a sparse array length.

## Suggested Fix

Promote before multiplying, or compare the element count against a divided limit:

```zig
const event_size = @sizeOf(Event);
if (@as(u64, array_length) > constants.message_body_size_max / event_size) {
    return request_error(env, .ERR_TOO_MUCH_DATA);
}
```

Add a Node regression test with a sparse array whose length exceeds the maximum request body size, expecting `ERR_TOO_MUCH_DATA` rather than a process abort.

--- a/src/clients/node/node.zig
+++ b/src/clients/node/node.zig
@@ -278,7 +278,7 @@ fn request(
             // Avoid allocating memory for requests that are known to be too large.
             // However, the final validation happens in `tb_client` against the runtime-known
             // maximum size.
-            if (array_length * @sizeOf(Event) > constants.message_body_size_max) {
+            if (array_length * @as(u64, @sizeOf(Event)) > constants.message_body_size_max) {
                 return request_error(env, .ERR_TOO_MUCH_DATA);
             }
 

--- a/src/clients/node/node.zig
+++ b/src/clients/node/node.zig
@@ -278,7 +278,10 @@ fn request(
             // Avoid allocating memory for requests that are known to be too large.
             // However, the final validation happens in `tb_client` against the runtime-known
             // maximum size.
-            if (array_length * @as(u64, @sizeOf(Event)) > constants.message_body_size_max) {
+            const event_max: u32 = comptime operation_comptime.event_max(
+                constants.message_body_size_max,
+            );
+            if (array_length > event_max) {
                 return request_error(env, .ERR_TOO_MUCH_DATA);
             }
 

--- a/src/clients/node/src/test.ts
+++ b/src/clients/node/src/test.ts
@@ -153,6 +153,17 @@ test('batch max size', async (): Promise<void> => {
   })
 })
 
+test('batch invalid size', async (): Promise<void> => {
+  const transfers: Transfer[] = [];
+  transfers.length = 0xffffffff;
+
+  assert.rejects(async() => await client.createTransfers(transfers), (err) => {
+    assert.ok(err instanceof RequestError)
+    assert.strictEqual(err.code,  ErrorCodes.ERR_TOO_MUCH_DATA)
+    return true
+  })
+})
+
 test('can lookup accounts', async (): Promise<void> => {
   const accounts = await client.lookupAccounts([accountA.id, accountB.id])
 

